lib/srv/db/cloud/iam_test.go

```diff
@@ -93,65 +101,112 @@ func TestAWSIAM(t *testing.T) {
 	}, types.DatabaseSpecV3{
 		Protocol: defaults.ProtocolPostgres,
 		URI:      "localhost",
-		AWS:      types.AWS{Redshift: types.Redshift{ClusterID: "redshift-cluster-1"}},
+		AWS:      types.AWS{Region: "localhost", AccountID: "1234567890", Redshift: types.Redshift{ClusterID: "redshift-cluster-1"}},
+	})
+	require.NoError(t, err)
+
+	databaseMissingMetadata, err := types.NewDatabaseV3(types.Metadata{
+		Name: "redshift",
+	}, types.DatabaseSpecV3{
+		Protocol: defaults.ProtocolPostgres,
+		URI:      "localhost",
+		AWS:      types.AWS{Redshift: types.Redshift{ClusterID: "missing metadata"}},
 	})
 	require.NoError(t, err)
 
 	// Make configurator.
+	taskChan := make(chan struct{})
+	waitForTaskProcessed := func(t *testing.T) {
+		select {
+		case <-taskChan:
+		case <-time.After(5 * time.Second):
+			require.Fail(t, "Failed to wait for task is processed")
+		}
+	}
 	configurator, err := NewIAM(ctx, IAMConfig{
+		AccessPoint: &mockAccessPoint{},
 		Clients: &common.TestCloudClients{
 			RDS:      rdsClient,
 			Redshift: redshiftClient,
 			STS:      stsClient,
 			IAM:      iamClient,
 		},
 		HostID: "host-id",
+		onProcessedTask: func(iamTask, error) {
+			taskChan <- struct{}{}
+		},
 	})
 	require.NoError(t, err)
+	require.NoError(t, configurator.Start(ctx))
 
-	// Configure RDS database and make sure IAM was enabled and policy was attached.
-	err = configurator.Setup(ctx, rdsDatabase)
-	require.NoError(t, err)
-	require.True(t, aws.BoolValue(rdsInstance.IAMDatabaseAuthenticationEnabled))
-	policy := iamClient.attachedRolePolicies["test-role"]["teleport-host-id"]
-	require.Contains(t, policy, rdsDatabase.GetAWS().RDS.ResourceID)
-
-	// Deconfigure RDS database, policy should get detached.
-	err = configurator.Teardown(ctx, rdsDatabase)
+	policyName, err := configurator.getPolicyName()
 	require.NoError(t, err)
-	policy = iamClient.attachedRolePolicies["test-role"]["teleport-host-id"]
-	require.NotContains(t, policy, rdsDatabase.GetAWS().RDS.ResourceID)
 
-	// Configure Aurora database and make sure IAM was enabled and policy was attached.
-	err = configurator.Setup(ctx, auroraDatabase)
-	require.NoError(t, err)
-	require.True(t, aws.BoolValue(auroraCluster.IAMDatabaseAuthenticationEnabled))
-	policy = iamClient.attachedRolePolicies["test-role"]["teleport-host-id"]
-	require.Contains(t, policy, auroraDatabase.GetAWS().RDS.ResourceID)
+	t.Run("RDS", func(t *testing.T) {
```

Comment:
Any chances to run all tests in parallel?

Reply:
i tried adding parallel but the test fails pretty bad as the test and the mock is not `-race` ready. I spent a little time but seems no easy fix without refactor the test so maybe we add parallel in a later time.